thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/DetectionConfigTuner.java

```diff
@@ -112,13 +120,14 @@ public DetectionConfigTuner(DetectionConfigDTO config, DataProvider dataProvider
     DateTime end =  new DateTime(endTime, timezone).withTimeAtStartOfDay();
     Interval window = new Interval(start, end);
 
-    // TODO can you use componentProps instead? what is the difference btw component and componentSpec
-    Map<String, Object> existingComponentSpec =
-        detectionConfig.getComponents().containsKey(componentKey) ?
-            MapUtils.getMap(detectionConfig.getComponents(), componentKey) : Collections.emptyMap();
-
     // TODO: if dimension drill down applied, pass in the metric urn of top dimension
-    return tunable.tune(existingComponentSpec, window, this.metricUrn);
+    tunedSpec.putAll(tunable.tune(componentProps, window, this.metricUrn));
+
+    // Hack to retain the raw yaml parameters.
+    // The tunable requires raw yaml params and previously tuned params to generate fresh params
+    tunedSpec.put(PROP_YAML_PARAMS, yamlParams);
```

Comment:
After tuning, do we need to retain the raw yaml parameters?

Reply:
That's a good question! I included it to keep it consistent with,
1. how we retain the raw params for other non-tunable components
2. how we include this in the spec before calling the tuner

Reply:
make sense. it can also be used in re-tune.